common/pde.h

```diff
@@ -0,0 +1,658 @@
+		return BufferWriter::Size(1 + 1 + 3 * (workers.size() + workers_gc.size()));
+	}
+
+	void WriteToBuffer(BufferWriter& writer)
+	{
+		writer.StartBlock("CommonData", Size());
+		writer.Write64(workers.size());
+		writer.Write64(workers_gc.size());
+		for (auto [core_id, worker_info] : workers)
+		{
+			writer.Write64(core_id);
+			writer.Write64(worker_info.socket_id);
+			writer.Write64(worker_info.start_at_memory);
+		}
+		for (auto [core_id, worker_info] : workers_gc)
+		{
+			writer.Write64(core_id);
+			writer.Write64(worker_info.socket_id);
+			writer.Write64(worker_info.start_at_memory);
+		}
+	}
+	void ReadFromBuffer(BufferReader& reader)
+	{
+		reader.StartBlock("CommonData");
+		uint64_t count_workers = reader.Read64();
+		uint64_t count_workers_gc = reader.Read64();
+
+		for (uint64_t index = 0; index < count_workers; ++index)
+		{
+			uint64_t core_id = reader.Read64();
+			tSocketId socket_id = reader.Read64();
+			uint64_t start_at_memory = reader.Read64();
+			workers[core_id] = {socket_id, start_at_memory, nullptr};
+			sockets[socket_id] = nullptr;
+		}
+
+		for (uint64_t index = 0; index < count_workers_gc; ++index)
+		{
+			uint64_t core_id = reader.Read64();
+			tSocketId socket_id = reader.Read64();
+			uint64_t start_at_memory = reader.Read64();
+			workers_gc[core_id] = {socket_id, start_at_memory, nullptr};
+			sockets[socket_id] = nullptr;
+		}
+	}
+
+	eResult FillWorkers(const std::map<tCoreId, tSocketId>& workers_to_sockets,
+	                    const std::map<tCoreId, tSocketId>& workers_gc_to_sockets,
+	                    size_t size_worker,
+	                    size_t size_worker_gc,
+	                    bool useHugeMem)
+	{
+		// Calculate sizes for each socket
+		std::map<tSocketId, size_t> sizes_by_sockets;
+		for (auto [core_id, socket_id] : workers_to_sockets)
+		{
+			workers[core_id] = {socket_id, sizes_by_sockets[socket_id], nullptr};
+			sizes_by_sockets[socket_id] += size_worker;
+		}
+		for (auto [core_id, socket_id] : workers_gc_to_sockets)
+		{
+			workers_gc[core_id] = {socket_id, sizes_by_sockets[socket_id], nullptr};
+			sizes_by_sockets[socket_id] += size_worker_gc;
+		}
+
+		// Create shared memory files by sockets
+		for (auto [socket_id, size] : sizes_by_sockets)
+		{
+			std::string fname = GetNameOfSocketFileName(socket_id);
+			void* buffer = SharedMemory::CreateBufferInSharedMemory(fname.c_str(), size, useHugeMem, true, socket_id);
+			if (buffer == nullptr)
+			{
+				return eResult::errorInitSharedMemory;
+			}
+			sockets[socket_id] = buffer;
+		}
+
+		// Set buffers for workers
+		for (auto [core_id, socket_id] : workers_to_sockets)
+		{
+			workers[core_id].buffer = PtrAdd(sockets[socket_id], workers[core_id].start_at_memory);
+		}
+		for (auto [core_id, socket_id] : workers_gc_to_sockets)
+		{
+			workers_gc[core_id].buffer = PtrAdd(sockets[socket_id], workers_gc[core_id].start_at_memory);
+		}
+
+		return eResult::success;
+	}
+};
+
+struct MainFileData
```

Comment:
I feel the name could be misleading. The PR suggests that only `DataPlane` manages counters in shared memory what could not be true in the future.